x/wasm/simulation/operations.go

```diff
@@ -110,6 +133,244 @@ func WeightedOperations(
 				DefaultSimulationExecutePayloader,
 			),
 		),
+		simulation.NewWeightedOperation(
+			weightMsgUpdateAdmin,
+			SimulateMsgUpdateAmin(
+				ak,
+				bk,
+				wasmKeeper,
+				DefaultSimulationUpdateAdminContractSelector,
+			),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgClearAdmin,
+			SimulateMsgClearAdmin(
+				ak,
+				bk,
+				wasmKeeper,
+				DefaultSimulationClearAdminContractSelector,
+			),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgMigrateContract,
+			SimulateMsgMigrateContract(
+				ak,
+				bk,
+				wasmKeeper,
+				DefaultSimulationMigrateContractSelector,
+				DefaultSimulationMigrateCodeIDSelector,
+			),
+		),
+	}
+}
+
+type MsgMigrateContractSelector func(sdk.Context, WasmKeeper, string) (sdk.AccAddress, types.ContractInfo)
+type MsgMigrateCodeIDSelector func(sdk.Context, WasmKeeper, uint64) uint64
+
+func DefaultSimulationMigrateContractSelector(ctx sdk.Context, wasmKeeper WasmKeeper, adminAddress string) (sdk.AccAddress, types.ContractInfo) {
+	var contractAddress sdk.AccAddress
+	var contractInfo types.ContractInfo
+	wasmKeeper.IterateContractInfo(ctx, func(address sdk.AccAddress, info types.ContractInfo) bool {
+		if info.Admin != adminAddress {
+			return false
+		}
+		contractAddress = address
+		contractInfo = info
+		return true
+	})
+	return contractAddress, contractInfo
+}
+
+func DefaultSimulationMigrateCodeIDSelector(ctx sdk.Context, wasmKeeper WasmKeeper, currentCodeID uint64) uint64 {
+	var codeID uint64
+	wasmKeeper.IterateCodeInfos(ctx, func(u uint64, info types.CodeInfo) bool {
+		if (info.InstantiateConfig.Permission != types.AccessTypeEverybody) || (u == currentCodeID) {
+			return false
+		}
+		codeID = u
+		return true
+	})
+	return codeID
+}
+
+func SimulateMsgMigrateContract(
+	ak types.AccountKeeper,
+	bk BankKeeper,
+	wasmKeeper WasmKeeper,
+	contractSelector MsgMigrateContractSelector,
+	codeIDSelector MsgMigrateCodeIDSelector,
+) simtypes.Operation {
+	return func(
+		r *rand.Rand,
+		app *baseapp.BaseApp,
+		ctx sdk.Context,
+		accs []simtypes.Account,
+		chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		simAccount, _ := simtypes.RandomAcc(r, accs)
+		ctAddress, _ := contractSelector(ctx, wasmKeeper, simAccount.Address.String())
+		if ctAddress == nil {
+			return simtypes.NoOpMsg(types.ModuleName, types.MsgMigrateContract{}.Type(), "no contract instance available"), nil, nil
+		}
+
+		var wasmBz []byte
+		var err error
+		wasmBz, err = os.ReadFile("./../x/wasm/keeper/testdata/reflect_1_1.wasm")
```

Comment:
I tried to avoid hard coding any contracts here. See L106-7. 
As we only store reflect contracts you can just pick a different code id and drop the store code part.

Reply:
I have fixed to embed